### PR TITLE
Add TDD-style practical materials

### PR DESCRIPTION
**Pull request by TheoLeanse**

### CODE REVIEW

**Comment by TheoLeanse** [comment]
@wheresrhys @jevakallio @jamesseanwright @kkarimi

**Comment by TheoLeanse** [comment]
WIP

- [ ] extend session plan into a tutorial-style resource that can guide the pairs through the practical session and help them make sense of the failing tests
- [ ] confirm what we want the API to do, and translate that into the test suite. Currently tests try to ensure that the api has routes for all codes (`/api/code`) and also has a route to accept a status code as a URL parameter (`/api/code/200` or `/api/code/404` etc) and return info about just that one code
- [ ] how to test the HTML app's behaviour

File: week-2-html/test.spec.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			.get('/600')
+			.expect(404)
+			.expect(res => {
+				expect(stubs['node-fetch']).calledWith('http://localhost:8080/api/code/600');
+			});
+	});
+
+
+	xit('renders mustache (?) templates that live in the views directory, and correctly renders the data into the view, too', () => {
+		// not sure how, or whether, to test this
```
Comment (TheoLeanse) [inline week-2-html/test.spec.js:69]: @jevakallio any ideas?

Comment (jamesseanwright) [reply week-2-html/test.spec.js:69]: That name suggests the test should assert that the templating engine should render something as expected. I don't think there's any value in doing that, as one would hope that the Handlebars guys are correctly testing their own code!

To start with, I'd write functional tests for the Express app; this is easily achievable with [SuperTest](https://github.com/visionmedia/supertest):

```
'use strict';

const request = require('supertest');
const myApp = require('./myExpressApp');

request(app)
    .get('/my-endpoint')
    .expect('Content-Type', /html/)
    .expect(200);
```
If you want to assert the returned markup then you could use [jsdom](https://github.com/tmpvar/jsdom) or [cheerio](https://github.com/cheeriojs/cheerio) to create a DOM from the response's body.

Comment (TheoLeanse) [reply week-2-html/test.spec.js:69]: yep - @jevakallio sorted that out! https://github.com/Code-Your-Future/node-status-code-website/blob/master/test/server.spec.js

**Pull request closed without merging**